Tasks/ServiceFabricDeploy/utilities.ps1

```diff
@@ -0,0 +1,123 @@
+    {
+        throw (Get-VstsLocString -Key ItemSearchNoFilesFound -ArgumentList $Pattern) 
+    }
+}
+
+# Function that can be mocked by tests
+function Create-Object
+{
+    Param (
+        [String]
+        $TypeName,
+
+        [Object[]]
+        $ArgumentList
+    )
+
+    return New-Object -TypeName $TypeName -ArgumentList $ArgumentList
+}
+
+function Get-AadSecurityToken
+{
+    Param (
+        [Hashtable]
+        $ClusterConnectionParameters,
+        
+        $ConnectedServiceEndpoint
+    )
+    
+    # Configure connection parameters to get cluster metadata
+    $connectionParametersWithGetMetadata = $ClusterConnectionParameters.Clone()
+    $connectionParametersWithGetMetadata.Add("GetMetadata", $true)
+    
+    # Query cluster metadata
+    $connectResult = Connect-ServiceFabricCluster @connectionParametersWithGetMetadata
+    $authority = $connectResult.AzureActiveDirectoryMetadata.Authority
+    Write-Host (Get-VstsLocString -Key AadAuthority -ArgumentList $authority)
+    $clusterApplicationId = $connectResult.AzureActiveDirectoryMetadata.ClusterApplication
+    Write-Host (Get-VstsLocString -Key ClusterAppId -ArgumentList $clusterApplicationId)
+    $clientApplicationId = $connectResult.AzureActiveDirectoryMetadata.ClientApplication
+    Write-Host (Get-VstsLocString -Key ClientAppId -ArgumentList $clientApplicationId)
+
+    # Acquire AAD access token
+    $serverOMDirectory = Get-VstsTaskVariable -Name 'Agent.ServerOMDirectory' -Require
+	Add-Type -LiteralPath "$serverOMDirectory\Microsoft.IdentityModel.Clients.ActiveDirectory.dll"	
+	$authContext = Create-Object -TypeName Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext -ArgumentList @($authority)
+    $authParams = $ConnectedServiceEndpoint.Auth.Parameters
+	$userCredential = Create-Object -TypeName Microsoft.IdentityModel.Clients.ActiveDirectory.UserCredential -ArgumentList @($authParams.Username, $authParams.Password)
+    
+    # Acquiring a token using UserCredential implies a non-interactive flow. No credential prompts will occur.
+    $accessToken = $authContext.AcquireToken($clusterApplicationId, $clientApplicationId, $userCredential).AccessToken
+
+    return $accessToken
+}
+
+function Read-XmlElementAsHashtable
+{
+    Param (
+        [System.Xml.XmlElement]
+        $Element
+    )
+
+    $hashtable = @{}
+    if ($Element.Attributes)
+    {
+        $Element.Attributes | 
+            ForEach-Object {
+                # Only boolean values are strongly-typed.  All other values are treated as strings.
+                $boolVal = $null
+                if ([bool]::TryParse($_.Value, [ref]$boolVal)) {
+                    $hashtable[$_.Name] = $boolVal
+                }
+                else {
+                    $hashtable[$_.Name] = $_.Value
+                }
+            }
+    }
+
+    return $hashtable
+}
+
+function Read-PublishProfile
+{
+    Param (
+        [ValidateScript({Test-Path $_ -PathType Leaf})]
```

Comment:
Does this syntax work in PS 3? I know before 5 (or 4?), creating a scriptblock required more cumbersome syntax (e.g. `[scriptblock]::create('string')` versus `{ some cmd }`. Although I'm not sure about within ValidateScript specifically.